Replace MockUsage with Usage.test in build tests

jmagman opened the pull request:
## Description

Leverage `AnalyticsMock` logging feature to capture usage events from `Usage.test()` by redirecting its console `print` to a string buffer.  Replace a few `MockUsage` instances.

## Related Issues

https://github.com/flutter/flutter/pull/67598#discussion_r501899905

jmagman committed:
Replace MockUsage with Usage.test in build tests

jonahwilliams reviewed (commented):
common.dart would be a good place for a helper to wrap the zone logic:

```
Future<void> runWithLogger(Future<void> Function() cb, Logger logger) {
  return runZoned(cb, zoneSpecification ...);
}
```

jmagman committed:
Use StringBuffer

jmagman commented:
> common.dart would be a good place for a helper to wrap the zone logic:
> 
> ```
> Future<void> runWithLogger(Future<void> Function() cb, Logger logger) {
>   return runZoned(cb, zoneSpecification ...);
> }
> ```

Added helper.  Also changed it to a `StringBuffer` and migrated `daemon_test` to use the helper.

jonahwilliams reviewed (approved):
LGTM

jmagman committed:
Make Usage.test() testWithoutContext safe

jmagman committed:
remove regex

jmagman committed:
Remove unused constructor param

jmagman committed:
Revert "remove regex"

This reverts commit 63b4c50815b1bdff003afe936d69d8b64e8dc68b.

jmagman committed:
Fix test
